web/src/main/java/com/bazaarvoice/emodb/web/scanner/ScanUploader.java

```diff
@@ -96,33 +115,60 @@ private ScanPlan createPlan(String scanId, ScanOptions options) {
     }
 
     private void startScanUpload(String scanId, ScanStatus status) {
-        boolean scanCreated = false;
+        final AtomicReference<Boolean> scanCreated = new AtomicReference<>(false);
 
         try {
-            // Create the scan
-            _scanStatusDAO.updateScanStatus(status);
-            scanCreated = true;
+            // compaction control timestamp = stash start time + 1 minute buffer time. This is needed to allow the setting time to trickle the request to the DataStore.
+            // Setting the time in the future takes care of the issue of there being any in-flight compactions
+            // Note: the same compaction control timestamp with 1 minute buffer time is also considered during the multiscan deltas/compactions resolving.
+            long compactionControlTime = status.getCompactionControlTime().getTime();
+            // expired time for now is designed to be 10 hours from the compaction control time.
+            long expireTime = compactionControlTime + Duration.ofHours(10).toMillis();
+            // Update the scan start time in Zookeeper in all data centers.
+            _compactionControlSource.updateStashTime(scanId, compactionControlTime, Lists.newArrayList(status.getOptions().getPlacements()), expireTime, _dataCenters.getSelf().getName());
+        } catch (Exception e) {
+            _log.error("Failed to update the stash time for scan {}", scanId, e);
+            throw Throwables.propagate(e);
+        }
 
-            // Notify the workflow that the scan can be started
-            _scanWorkflow.scanStatusUpdated(scanId);
+        // spawn a thread and do the below asynchronously as we would have to wait for a minute to continue to scan and we don't to include that delay in here for responding.
+        new Thread(() ->
+        {
+            try {
+                // We would like to wait 1 minute here to continue to scan to make sure scan don't miss any deltas that are written before the compaction control time.
+                Thread.sleep(_compactionControlBufferTimeInMillis);
```

Comment:
This sleep is necessary, but doing it here seems dangerous.  If there is a failure, such as process death while sleeping, there is nothing to ensure the scan starts.  Maybe it makes more sense to sleep after the first message is pulled from the queue.  That way if there is a failure the queued message will eventually be redelivered.